types/rosie/index.d.ts

```diff
@@ -187,9 +210,9 @@ declare namespace rosie {
      * @param {object=} options
      * @return {*}
      */
-    build(attributes: Object, options: Object): Object;
+    build(attributes: { [k in keyof T]?: T[k] }, options?: any): T;
```

Comment:
I can see now that only the keys are optional, not the entire parameter. Would you like to cut a new PR or should I? :)

```ts
build(attributes: { [k in keyof T]?: T[k] }?, options?: any): T;
```

I believe that is the correct syntax